ast/print.go

```diff
@@ -45,21 +47,28 @@ func (n *ConstantNode) String() string {
 }
 
 func (n *UnaryNode) String() string {
-	op := ""
+	if EnableCache && n.strCache != "" {
```

Comment:
I checkout a new branch [feature-cache_print_vs](https://github.com/zhuliquan/expr/tree/feature-cache_print_vs) from master, and I compare two branch benchmark, as below: 
```shell
# zlq16 at XIAOZHUPC in D:\workspace\go\src\github.com\zhuliquan\expr on git:feature-cache_print_vs ≢ +2 ~0 -0 ! inside env:base [21:43:00]
➜ go test -benchmem -run=^$ -bench ^BenchmarkCompileWithProfile$ github.com/expr-lang/expr/compiler -cpuprofile=profile1.txt -benchtime=10s
goos: windows
goarch: amd64
pkg: github.com/expr-lang/expr/compiler      
cpu: Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
BenchmarkCompileWithProfile-8               6010            194681 ns/op           91430 B/op       1286 allocs/op
PASS
ok      github.com/expr-lang/expr/compiler      3.115s
# zlq16 at XIAOZHUPC in D:\workspace\go\src\github.com\zhuliquan\expr on git:feature-cache_print_vs ≢ +2 ~0 -0 ! inside env:base [21:43:59]
➜ git branch
* feature-cache_print_vs
  feature-cahce_print
  feature-common_reused1
  feature-experiment
  master
# zlq16 at XIAOZHUPC in D:\workspace\go\src\github.com\zhuliquan\expr on git:feature-cache_print_vs ≢ +2 ~0 -0 ! inside env:base [21:44:05]
➜ git checkout feature-cahce_print
Switched to branch 'feature-cahce_print'
Your branch is ahead of 'origin/feature-cahce_print' by 2 commits.
  (use "git push" to publish your local commits)
# zlq16 at XIAOZHUPC in D:\workspace\go\src\github.com\zhuliquan\expr on git:feature-cahce_print ↑2 +2 ~0 -0 ! inside env:base 
[21:44:13]
➜ go test -benchmem -run=^$ -bench ^BenchmarkCompileWithProfile$ github.com/expr-lang/expr/compiler -cpuprofile=profile2.txt -benchtime=10s
goos: windows
goarch: amd64
pkg: github.com/expr-lang/expr/compiler      
cpu: Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
BenchmarkCompileWithProfile-8               7676            137161 ns/op           83643 B/op        638 allocs/op
PASS
ok      github.com/expr-lang/expr/compiler      2.067s
```
and I got two profile file, profile1.txt for with no cache, and profile2.txt for cache printing. (you can run `go tool pprofile --http 127.0.0.1:8777 profile1.txt` or  `go tool pprofile --http 127.0.0.1:8778 profile2.txt` to got flame graph)  
[profile1.txt](https://github.com/expr-lang/expr/files/14893769/profile1.txt)
[profile2.txt](https://github.com/expr-lang/expr/files/14893770/profile2.txt)

According to profile, in case of cache printing, system invoke `String()` fewer times than no cache.